### PR TITLE
feat(monitoring): Refactor scheduler to monitoring and reformat tasks…

### PR DESCRIPTION
**Pull request by cassand**
… and query models

### CODE REVIEW

File: backend/app/controllers/SchedulerController.scala [hunk 1]
```diff
@@ -32,8 +35,10 @@ class SchedulerController @Inject() (schedulerService: SchedulerService) extends
     *
```
Comment (FunkySayu) [inline backend/app/controllers/SchedulerController.scala:35]: Add function description

File: backend/app/models/Query.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package models
+
+import play.api.libs.json._
+
+
```
Comment (FunkySayu) [inline backend/app/models/Query.scala:5]: Remove blank line (or add another for consistency in other files after the imports)

**Comment by ArchangelX360** [comment]
LGTM.

File: backend/app/modules/scheduler/MonitoringActor.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    * Create a new instance of monitoring actor
+    */
+  def props = Props[MonitoringActor]
+
+  /**
+    * Message definition
+    */
+  case class StartProcess(query: String)
+  case class UpdateProcess(id: String, status: String)
+
```
Comment (FunkySayu) [inline backend/app/modules/scheduler/MonitoringActor.scala:20]: Remove blank line if this is part of the 'Message definition group' (or add a comment on the ListProcess, but I don't think it's the case)

File: backend/app/modules/scheduler/MonitoringActor.scala [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  /**
+    * Implicit import from companion object
+    */
+  import MonitoringActor._
+
+  /**
+    * List of all the process run by the server (including finished ones)
+    */
+  val processList = mutable.MutableList.empty[Query]
```
Comment (FunkySayu) [inline backend/app/modules/scheduler/MonitoringActor.scala:34]: Isn't this possibly dangerous? I mean if the number of query grow, the list of process will also grow a lot and eventually consume too much memory.

File: backend/app/modules/scheduler/MonitoringActor.scala [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    case StartProcess(query) => sender ! startProcess(query)
+    case ListProcess(status) => sender ! listProcess(status)
+    case _ => ()
+  }
+
+  /**
+    * Create monitoring for a new process/query
+    */
+  def startProcess(query: String): Query = {
+    val process = Query(
```
Comment (FunkySayu) [inline backend/app/modules/scheduler/MonitoringActor.scala:49]: `val process =` -> `processList +=` and remove the append below (+ the blank line)

File: backend/app/modules/scheduler/MonitoringActor.scala [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  override def receive: Receive = {
+    case StartProcess(query) => sender ! startProcess(query)
+    case ListProcess(status) => sender ! listProcess(status)
+    case _ => ()
+  }
+
+  /**
+    * Create monitoring for a new process/query
+    */
+  def startProcess(query: String): Query = {
```
Comment (FunkySayu) [inline backend/app/modules/scheduler/MonitoringActor.scala:48]: `query` not in the function doc

File: backend/app/modules/searchengine/SearchActor.scala [hunk 7]
```diff
@@ -36,16 +39,16 @@ class SearchActor (scheduler: ActorRef) extends Actor {
 
   /**
     * Handle search message, send it to spotlight for TAL and contact other actors to schedule the processing
+ *
```
Comment (FunkySayu) [inline backend/app/modules/searchengine/SearchActor.scala:42]: delete this line for consistency

File: backend/app/modules/searchengine/SearchActor.scala [hunk 8]
```diff
@@ -36,16 +39,16 @@ class SearchActor (scheduler: ActorRef) extends Actor {
...
+  def processMessage(message: String): Query = {
+    val schedulerResponse = (scheduler ? MonitoringActor.StartProcess(message)).mapTo[Query]
 
-    // We need information on the process to go further, therefore we wait for the process creation
+    // We need information on the process to go further, therefore we wait for the process to be initialised
     val process = Await.result(schedulerResponse, Duration.Inf)
 
-
     process
   }
```
Comment (FunkySayu) [inline backend/app/modules/searchengine/SearchActor.scala:53]: + blank line after for consistency ?

File: backend/app/modules/searchengine/SearchActor.scala [hunk 9]
```diff
@@ -3,16 +3,13 @@ package modules.searchengine
...
+import models.Query
+import modules.scheduler.MonitoringActor
 
 import scala.concurrent.Await
-import scala.concurrent.Future
-import scala.concurrent.duration.Duration
-import scala.concurrent.duration._
-
-import models.Process
+import scala.concurrent.duration.{Duration, _}
```
Comment (FunkySayu) [inline backend/app/modules/searchengine/SearchActor.scala:12]: What's the point of importing *Duration* and *everything* afterward? Isn't Duration part of everything?

**Pull request merged**